### ISSUE TITLE
listing all pods leads to "sources must not be None" error in v1_projected_volume_source.py

### ISSUE THREAD

**Issue by zapman449** [kubernetes-client/python#1494]
**What happened (please include outputs or screenshots)**:
```
Traceback (most recent call last):
  File "/opt/autoscaler/ski_autoscaler/lifecycle_manager.py", line 566, in lifecycle_manager
    main(
  File "/opt/autoscaler/ski_autoscaler/lifecycle_manager.py", line 478, in main
    asg_pods = kube_filter.pods_in_asg_filter(kube_filter.get_all_pods(), asg_nodes)
  File "/opt/autoscaler/ski_utils/kube_filter.py", line 86, in get_all_pods
    result = v1.list_pod_for_all_namespaces(watch=False)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api/core_v1_api.py", line 14098, in list_pod_for_all_namespaces
    (data) = self.list_pod_for_all_namespaces_with_http_info(**kwargs)  # noqa: E501
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api/core_v1_api.py", line 14179, in list_pod_for_all_namespaces_with_http_info
    return self.api_client.call_api(
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 340, in call_api
    return self.__call_api(resource_path, method,
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 184, in __call_api
    return_data = self.deserialize(response_data, response_type)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 256, in deserialize
    return self.__deserialize(data, response_type)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 295, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 636, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 272, in __deserialize
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 272, in <listcomp>
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 295, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 636, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 295, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 636, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 272, in __deserialize
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 272, in <listcomp>
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 295, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 636, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 295, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/api_client.py", line 638, in __deserialize_model
    instance = klass(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/models/v1_projected_volume_source.py", line 52, in __init__
    self.sources = sources
  File "/usr/local/lib/python3.8/site-packages/kubernetes/client/models/v1_projected_volume_source.py", line 98, in sources
    raise ValueError("Invalid value for `sources`, must not be `None`")  # noqa: E501
ValueError: Invalid value for `sources`, must not be `None`
```

**What you expected to happen**:
return a list of all pods

**How to reproduce it (as minimally and precisely as possible)**:
One of my clusters does it every time, all the time, starting today. My other clusters do NOT exhibit the same behavior

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (`kubectl version`): v1.17.12-eks-7684af
- OS (e.g., MacOS 10.13.6): ubuntu 20.04
- Python version (`python --version`) 3.8 and 3.9
- Python client version (`pip list | grep kubernetes`) 11.0.0 and 17.17.0

**Comment by zapman449** [comment]
FWIW: `kubectl get pods --all-namespaces` functions as expected.

**Comment by zapman449** [comment]
The above traceback comes from v11 of the client

The following comes from v17.17:
```
Traceback (most recent call last):
  File "/opt/autoscaler/ski_autoscaler/lifecycle_manager.py", line 566, in lifecycle_manager
    main(
  File "/opt/autoscaler/ski_autoscaler/lifecycle_manager.py", line 478, in main
    asg_pods = kube_filter.pods_in_asg_filter(kube_filter.get_all_pods(), asg_nodes)
  File "/opt/autoscaler/ski_utils/kube_filter.py", line 86, in get_all_pods
    result = v1.list_pod_for_all_namespaces(watch=False)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 16864, in list_pod_for_all_namespaces
    return self.list_pod_for_all_namespaces_with_http_info(**kwargs)  # noqa: E501
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 16967, in list_pod_for_all_namespaces_with_http_info
    return self.api_client.call_api(
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 348, in call_api
    return self.__call_api(resource_path, method,
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 192, in __call_api
    return_data = self.deserialize(response_data, response_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 264, in deserialize
    return self.__deserialize(data, response_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in __deserialize
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in <listcomp>
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in __deserialize
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 280, in <listcomp>
    return [self.__deserialize(sub_data, sub_kls)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 639, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 303, in __deserialize
    return self.__deserialize_model(data, klass)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 641, in __deserialize_model
    instance = klass(**kwargs)
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/models/v1_projected_volume_source.py", line 57, in __init__
    self.sources = sources
  File "/usr/local/lib/python3.9/site-packages/kubernetes/client/models/v1_projected_volume_source.py", line 103, in sources
    raise ValueError("Invalid value for `sources`, must not be `None`")  # noqa: E501
ValueError: Invalid value for `sources`, must not be `None`
```

**Comment by zapman449** [comment]
Ahh.  Found a deployment -> replicaset -> pod a user had created with:

```yaml
       volumes:
       - name: secretsmount
         projected:
           defaultMode: 420
           sources: null
```

Kube let the pod in, python kubernetes SDK shouldn't blow up on this.

(working to prove this theory)

**Comment by zapman449** [comment]
Theory confirmed.  Removed that `volume` block (and the associated `volumeMount`) and the SDK was able to function normally.

So, steps to reproduce:
1. setup pod with a projected volume with `sources: null`
2. call `list_pod_for_all_namespaces(watch=False)`

**Comment by zapman449** [comment]
I presume a similar failure will occur for list_pods, list_replicasets, list_deployments, etc.

**Comment by roycaihw** [comment]
This was fixed in upstream in Kubernetes 1.20: https://github.com/kubernetes/kubernetes/pull/94161, but the Python client hasn't caught up with the Kubernetes releases (we are at 1.18). 

We can patch this client to cherrypick the fix. @zapman449 Would you like to open a PR to remove [these lines](https://github.com/kubernetes-client/python/blob/d8f283e7483848647804eab345645106b6fb357d/kubernetes/client/models/v1_projected_volume_source.py#L102-L104)? I believe that will fix this issue.
